tst/EnergyPlus/unit/SolarShading.unit.cc

```diff
@@ -1082,3 +1083,391 @@ TEST_F( EnergyPlusFixture, SolarShadingTest_ExternalShadingIO )
+		"    SunShading,              !- Transmittance Schedule Name                          ",
+		"    4,                       !- Number of Vertices                                   ",
+		"    -3,-5,2.5,  !- X,Y,Z ==> Vertex 1 {m}                                            ",
+		"    -3,-6,2.5,  !- X,Y,Z ==> Vertex 2 {m}                                            ",
+		"    3,-6,2.5,  !- X,Y,Z ==> Vertex 3 {m}                                             ",
+		"    3,-5,2.5;  !- X,Y,Z ==> Vertex 4 {m}                                             ",
+		"  ShadingProperty:Reflectance,                                                       ",
+		"    Zn001:Wall-South:Shade001,  !- Shading Surface Name                              ",
+		"    0.2,                     !- Diffuse Solar Reflectance of Unglazed Part of Shading",
+		"    0.2;                     !- Diffuse Visible Reflectance of Unglazed Part of Shadi"
+	} );
+
+	ASSERT_FALSE( process_idf( idf_objects ) );
+
+	SolarShading::clear_state();
+
+	SimulationManager::GetProjectData();
+	bool FoundError = false;
+
+	HeatBalanceManager::GetProjectControlData( FoundError ); // read project control data
+	EXPECT_FALSE( FoundError ); // expect no errors
+
+	HeatBalanceManager::SetPreConstructionInputParameters();
+	ScheduleManager::ProcessScheduleInput(); // read schedules
+
+	HeatBalanceManager::GetMaterialData( FoundError );
+	EXPECT_FALSE( FoundError );
+
+	HeatBalanceManager::GetFrameAndDividerData( FoundError );
+	EXPECT_FALSE( FoundError );
+
+	HeatBalanceManager::GetConstructData( FoundError );
+	EXPECT_FALSE( FoundError );
+
+	HeatBalanceManager::GetZoneData( FoundError ); // Read Zone data from input file
+	EXPECT_FALSE( FoundError );
+
+	SurfaceGeometry::GetGeometryParameters( FoundError );
+	EXPECT_FALSE( FoundError );
+
+	SizingManager::GetZoneSizingInput();
+	
+
+	SurfaceGeometry::CosZoneRelNorth.allocate( 1 );
+	SurfaceGeometry::SinZoneRelNorth.allocate( 1 );
+
+	SurfaceGeometry::CosZoneRelNorth( 1 ) = std::cos( -Zone( 1 ).RelNorth * DegToRadians );
+	SurfaceGeometry::SinZoneRelNorth( 1 ) = std::sin( -Zone( 1 ).RelNorth * DegToRadians );
+	SurfaceGeometry::CosBldgRelNorth = 1.0;
+	SurfaceGeometry::SinBldgRelNorth = 0.0;
+
+	compare_err_stream( "" ); // just for debugging
+	SurfaceGeometry::SetupZoneGeometry( FoundError ); // this calls GetSurfaceData()
+	EXPECT_FALSE( FoundError );	
+
+	compare_err_stream( "" ); // just for debugging
+
+	SolarShading::GetShadowingInput();
+
+	for ( int SurfNum = 1; SurfNum <= TotSurfaces; SurfNum++ ) {
+		if ( Surface( SurfNum ).ExtBoundCond == 0 && Surface( SurfNum ).Zone != 0 ) {
+			int ZoneSize = Surface(SurfNum).DisabledShadowingZoneList.size();
+			EXPECT_EQ( 1, ZoneSize );
+			std::vector< int > DisabledZones = Surface( SurfNum ).DisabledShadowingZoneList;
+			for ( int i : DisabledZones ) {
+				EXPECT_EQ( 1, i );
+			}
+		}
+	}
+
+	SolarShading::clear_state();
```

Comment:
As long as SolarShading::clear_state() is called in the EnergyPlusFixture class, then calling it here is not necessary and I'd like to get rid of it to avoid confusion.